### DISCUSSION TITLE
No `just` language in `just` repo

### DISCUSSION THREAD

**Discussion by ZeyadMoustafaKamal** [github-linguist/linguist#6744] [Classification] [answered]
## Describe the bug

So in `just` repo here https://github.com/casey/just/ I found that there is `Makefile` language not `just` for the `justfile` the thing is that in another repo like [this one](https://github.com/NoahTheDuke/vim-just) I can see `just` language appear their which means that `just` language have already been added here.

### Expected behaviour
To see the `just` language instead of `Makefile`

### Related discussion
Nothing
### Additional notes
Nothing

**Comment by lildude** [comment, marked as answer]
There are two things that explain this:

1. Just is the smallest language in that repo by bytes of code as Linguist currently detects things and the sidebar only shows the first 6 languages of a repo:

```console
➜ github-linguist just/
95.38%  644554     Rust
3.54%   23931      Shell
0.71%   4799       Makefile
0.19%   1275       CSS
0.12%   787        HTML
0.06%   375        Nushell
0.01%   57         Just

➜ 
```

2. The `justfile` is being "misclassified" in that repo because the author told Linguist to consider the file as a Makefile right at the [end with these lines](https://github.com/casey/just/blob/b4f779e4fe4d5adb95b1a802549a09a7067b97cf/justfile#L227-L230).

In order for Just to appear in the sidebar, all the author needs to do is remove the `vim` and `mode` lines:

```console                                                                                                                                                                                       
➜ github-linguist just/
95.39%  644554     Rust
3.54%   23931      Shell
0.71%   4792       Just
0.19%   1275       CSS
0.12%   787        HTML
0.06%   375        Nushell

➜ 
```

**Reply by ZeyadMoustafaKamal** [reply]
Ok, thanks. I understand this now.
